evalml/data_checks/datetime_format_data_check.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+
+from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.utils import infer_feature_types
+
+
+class DateTimeFormatDataCheck(DataCheck):
+    """Checks if the datetime column has equally spaced intervals and is monotonically increasing or decreasing in order
+    to be supported by time series estimators.
+
+    Arguments:
+        datetime_column (str): The name of the datetime column. If the datetime values are in the index, then pass "index".
```

Comment:
What if the column doesnt have a string name? ie unamed dataframe, int col?

Reply:
The data check supports that as well, but I'll update the docstring and add a test!